src/menu.cpp

```diff
@@ -2421,36 +2734,69 @@ void Menu::ProcessEvents( const SystemEvents& events )
 	{
 		switch( event.type )
 		{
-		case SystemEvent::Type::Key:
-			if(
-				event.event.key.pressed &&
-				event.event.key.key_code == KeyCode::Escape )
-			{
-				if( current_menu_ != nullptr )
+			case SystemEvent::Type::MouseKey:
+				if( event.event.mouse_key.pressed )
 				{
-					current_menu_->PlayMenuSound( Sound::SoundId::MenuOn );
-					MenuBase* const new_menu= current_menu_->GetParent();
-					if( new_menu != current_menu_ )
+					switch( event.event.mouse_key.mouse_button )
 					{
-						if( new_menu != nullptr )
-							new_menu->OnActivated();
-						current_menu_= new_menu;
+						case SystemEvent::MouseKeyEvent::Button::Right:
+							if( current_menu_ != nullptr )
+							{
+								current_menu_->PlayMenuSound( Sound::SoundId::MenuOn );
+								MenuBase* const new_menu= current_menu_->GetParent();
+								if( new_menu != current_menu_ )
+								{
+									if( new_menu != nullptr )
+										new_menu->OnActivated();
+									current_menu_= new_menu;
+								}
+							}
+							else
+							{
+								if( !host_commands_.CurrentMap() )
```

Comment:
Avoid implicit casts to bool (for better readability and errors prevention). Compare pointers explicitly against `nullptr`.